staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -238,6 +238,35 @@ func DialerFor(transport http.RoundTripper) (DialFunc, error) {
 	}
 }
 
+// CloseIdleConnectionsFor close idles connections for the Transport.
+// If the Transport is wrapped it iterates over the wrapped round trippers
+// until it finds one that implemens the CloseIdleConnections method.
+// If the Transport does not have a CloseIdleConnections method
+// then this function does nothing.
+func CloseIdleConnectionsFor(transport http.RoundTripper) {
+	if transport == nil {
+		return
+	}
+
+	switch transport := transport.(type) {
+	case *http.Transport:
+		transport.CloseIdleConnections()
+	case RoundTripperWrapper:
+		type closeIdler interface {
+			CloseIdleConnections()
+		}
+		if tr, ok := transport.(closeIdler); ok {
+			tr.CloseIdleConnections()
+			return
+		}
+		CloseIdleConnectionsFor(transport.WrappedRoundTripper())
+	default:
+		klog.Warningf("Transport doesn't implement CloseIdleConnections method")
```

Comment:
prefer returning an error the caller can determine how to handle, rather than logging warnings inside a utility helper method

Reply:
then, I think that not returning nothing keeps the same behavior as the standard library

https://github.com/golang/go/blob/aa4e0f528e1e018e2847decb549cfc5ac07ecf20/src/net/http/client.go#L937-L951

and is consistent with the definition.